online-shopping-website/backend/src/routes/users.ts

```diff
@@ -67,8 +67,12 @@ userRouter.post('/update', async (req: Request, res: Response) => { // updates a
     if (usr === null || usr === undefined) {
       throw new Error(`Authentication is invalid`);
     }
+    const match = await bcrypt.compare(req.body.oldPassword, usr.password);
+    if (!match){
+      throw new Error(`Password is incorrect`);
+    }
     let encrypted_password: string | undefined;
-    if (req.body.password !== undefined) {
+    if (req.body.newPassword !== undefined) {
```

Comment:
Would we be able to change this from `newPassword` back to `password`? This change would have effects elsewhere in the app because we are setting the cookie, and other pages look for `password` instead of `newPassword`. It's not a big deal to fix, just wanted to know if this is necessary.

Reply:
Yeah its possible but it might make things a little more difficult to read. If it makes it easier for the front end then i can switch it back